src/anonymeter/evaluators/inference_evaluator.py

```diff
@@ -14,30 +14,35 @@
 
 
 def _run_attack(
-        target: pd.DataFrame,
-        syn: pd.DataFrame,
-        n_attacks: int,
-        aux_cols: list[str],
-        secret: str,
-        n_jobs: int,
-        naive: bool,
-        regression: Optional[bool],
-        inference_model: Optional[InferencePredictor],
-) -> int:
+    target: pd.DataFrame,
+    syn: pd.DataFrame,
+    n_attacks: int,
+    aux_cols: list[str],
+    secret: str,
+    n_jobs: int,
+    naive: bool,
+    regression: Optional[bool],
+    inference_model: Optional[InferencePredictor],
+) -> tuple[int, pd.Series]:
     if regression is None:
         regression = pd.api.types.is_numeric_dtype(target[secret])
 
     targets = target.sample(n_attacks, replace=False)
 
     if naive:
         guesses = syn.sample(n_attacks)[secret]
+        guesses.index = targets.index
```

Comment:
NIT: use [`reindex_like`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.reindex_like.html#pandas.DataFrame.reindex_like):

```suggestion
        guesses.reindex_lie(targets)
```
(untested)